pkg/nexus/idle/scheduler/idle-scheduler.go

```diff
@@ -47,9 +45,10 @@ func (ds *IdleScheduler) executeSchedule() {
 
 		for ds.Queue.Len() > 0 && ds.MaxParallelRequests.Load() > 0 {
 			ds.MaxParallelRequests.Add(-1)
+			task := ds.Queue.Pop()
 			go func() {
 				defer ds.MaxParallelRequests.Add(1)
-				ds.Pop()
+				ds.CallSynchronized(task)
 			}()
```

Comment:
```go
task := ds.Queue.Pop()
			go func(taskInFunction *structs.NexusItem) {
				defer ds.MaxParallelRequests.Add(1)
				ds.Pop()
				ds.CallSynchronized(taskInFunction)
			}(task)
```


Damit du in go routine Fehler vermeidest solltest du die klammern der funktion nutzen.

Reply:
Done